app/products/calls/components/call_notification/call_notification.tsx

```diff
@@ -0,0 +1,227 @@
+        paddingRight: 12,
+        borderRadius: 8,
+        borderWidth: 2,
+        borderStyle: 'solid',
+        borderColor: changeOpacity(Preferences.THEMES.denim.buttonColor, 0.16),
+        gap: 8,
+        alignItems: 'center',
+        backgroundColor: changeOpacity('#000', 0.16),
+    },
+    innerOnCallsScreen: {
+        borderColor: changeOpacity(Preferences.THEMES.denim.buttonColor, 0.16),
+        backgroundColor: changeOpacity('#000', 0.12),
+    },
+    text: {
+        flex: 1,
+        ...typography('Body', 200),
+        lineHeight: 20,
+        color: Preferences.THEMES.denim.buttonColor,
+    },
+    boldText: {
+        ...typography('Body', 200, 'SemiBold'),
+        lineHeight: 20,
+    },
+    join: {
+        flexDirection: 'row',
+        alignItems: 'flex-end',
+        height: 40,
+        gap: 7,
+        backgroundColor: Preferences.THEMES.denim.buttonColor,
+        paddingTop: 10,
+        paddingRight: 20,
+        paddingBottom: 10,
+        paddingLeft: 20,
+        borderRadius: 30,
+    },
+    joinOnCallsScreen: {
+        backgroundColor: changeOpacity(Preferences.THEMES.denim.buttonColor, 0.12),
+    },
+    joinLabel: {
+        ...typography('Body', 100, 'SemiBold'),
+    },
+    joinIconLabel: {
+        color: Preferences.THEMES.denim.onlineIndicator,
+    },
+    joinIconLabelOnCallsScreen: {
+        color: Preferences.THEMES.denim.buttonColor,
+    },
+    dismiss: {
+        height: 40,
+        width: 40,
+        borderRadius: 20,
+        padding: 0,
+        backgroundColor: changeOpacity(Preferences.THEMES.denim.buttonColor, 0.08),
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    dismissOnCallsScreen: {
+        backgroundColor: 'transparent',
+    },
+    dismissIcon: {
+        color: Preferences.THEMES.denim.buttonColor,
+    },
+});
+
+type Props = {
+    incomingCall: IncomingCallNotification;
+    currentUserId: string;
+    teammateNameDisplay: string;
+    members?: ChannelMembershipModel[];
+    onCallsScreen?: boolean;
+}
+
+export const CallNotification = ({
+    incomingCall,
+    currentUserId,
+    teammateNameDisplay,
+    members,
+    onCallsScreen,
+}: Props) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+
+    useEffect(() => {
+        const channelMembers = members?.filter((m) => m.userId !== currentUserId);
+        if (!channelMembers?.length) {
+            fetchProfilesInChannel(serverUrl, incomingCall.channelID, currentUserId, undefined, false);
+        }
+    }, []);
+
+    const onJoinPress = useCallback(() => {
+        leaveAndJoinWithAlert(intl, incomingCall.serverUrl, incomingCall.channelID);
+    }, [intl, incomingCall]);
+
+    const onContainerPress = useCallback(async () => {
+        if (serverUrl !== incomingCall.serverUrl) {
+            await DatabaseManager.setActiveServerDatabase(incomingCall.serverUrl);
+            await WebsocketManager.initializeClient(incomingCall.serverUrl);
+        }
```

Comment:
Is there any special indication UX wise if a call is coming from a different server?

Reply:
No, not at the moment. We would have to add that to the designs. Should we add a ticket?

Reply:
Yeah, I remember a whole discussion over this (regarding Desktop). It can easily get confusing otherwise.

Reply:
https://mattermost.atlassian.net/browse/MM-54323